cmd/pod-scaler/frontend/src/app/CircLLHist/LogarithmicComparativePlot.tsx

```diff
@@ -0,0 +1,403 @@
+    handleResize();
+    return () => {
+      window.removeEventListener('resize', handleResize);
+    };
+  }, [container, props]);
+
+  const [first, setFirst] = React.useState<number>(0);
+
+  const onChange = (value: number): void => {
+    setFirst(value);
+  };
+
+  const [index, setIndex] = React.useState<Record<string, string>>({});
+  const mainCanvas: React.RefObject<HTMLCanvasElement> = React.useRef<HTMLCanvasElement>(null);
+  const hiddenCanvas: React.RefObject<HTMLCanvasElement> = React.useRef<HTMLCanvasElement>(null);
+  const tooltip: React.RefObject<HTMLDivElement> = React.useRef<HTMLDivElement>(null);
+
+  // recompute meta when data changes
+  React.useEffect(() => {
+    if (container.current) {
+      const width = container.current.getBoundingClientRect().width;
+      setCanvasMeta(canvasMetaFromProps(props, width))
+    }
+  }, [props])
+
+  // draw the canvases whenever we change the data or bounds
+  React.useEffect(() => {
+    if (mainCanvas.current && hiddenCanvas.current) {
+      setIndex(draw(canvasMeta, props.cutoff, props.merged, props.histograms.slice(first, first + canvasMeta.numCols - 1), mainCanvas.current, hiddenCanvas.current));
+    }
+  }, [canvasMeta, props.cutoff, props.merged, props.histograms, first])
+
+  React.useEffect(() => {
+    const handleMouseMove = (event: MouseEvent): void => {
+      if (mainCanvas.current && hiddenCanvas.current && tooltip.current) {
+        const context = hiddenCanvas.current.getContext("2d");
+        if (!context) {
+          return;
+        }
+        const color = context.getImageData(event.offsetX, event.offsetY, 1, 1).data;
+        const key = "rgb(" + color[0] + "," + color[1] + "," + color[2] + ")";
+        const rect = mainCanvas.current.getBoundingClientRect();
+        if (key in index) {
+          tooltip.current.style.opacity = "0.8";
+          tooltip.current.style.top = rect.y + event.offsetY + 5 + "px";
+          tooltip.current.style.left = rect.x + event.offsetX + 5 + "px";
+          tooltip.current.innerHTML = index[key];
+        } else {
+          tooltip.current.style.opacity = "0";
+        }
+      }
+    };
+    if (mainCanvas.current) {
+      const node = mainCanvas.current;
+      node.addEventListener('mousemove', handleMouseMove);
+      return () => {
+        node.removeEventListener('mousemove', handleMouseMove);
+      };
+    }
+    return;
+  }, [index, hiddenCanvas, mainCanvas, tooltip]);
+
+  const request: string = canvasMeta.yAxisFormatter(props.cutoff);
+  let samples = 0;
+  for (const bin of props.merged.bins) {
+    samples += bin.count;
+  }
+  const description: string = "Analyzing " + samples + " samples over " + props.histograms.length + " traces, a request of " + request + canvasMeta.yAxisUnit + " is recommended.";
+
+  let sliderMax: number = props.histograms.length - canvasMeta.numCols + 1;
+  if (sliderMax <= 0) {
+    sliderMax = 0;
+  }
+
+  return <Flex direction={{default: 'column'}}
```

Comment:
I personally think it's cleaner to use `div`s with CSS classes rather than the PF `Flex` components (separation of concerns).